main/src/ca/uwaterloo/flix/language/phase/unification/CaseSetUnification.scala

```diff
@@ -235,12 +235,12 @@ object CaseSetUnification {
   private def nnf(t: SetFormula)(implicit univ: Set[Int]): Nnf = t match {
     case Cst(syms) =>
       val lits: Set[Nnf] = syms.map(sym => Nnf.Singleton(Literal.Positive(Atom.Case(sym))))
-      lits.reduceLeftOption(Nnf.Union).getOrElse(Nnf.Empty)
+      lits.reduceLeftOption(Nnf.Union.apply).getOrElse(Nnf.Empty)
     case Var(sym) => Nnf.Singleton(Literal.Positive(Atom.Var(sym)))
     case Not(tpe) => nnfNot(tpe)
     case Or(tpe1, tpe2) => Nnf.Union(nnf(tpe1), nnf(tpe2))
     case And(tpe1, tpe2) => Nnf.Intersection(nnf(tpe1), nnf(tpe2))
-    case _ => throw InternalCompilerException(s"unexpected type: $t", SourceLocation.Unknown)
+    case null => throw InternalCompilerException(s"unexpected type: $t", SourceLocation.Unknown)
```

Comment:
This does not make sense to me. Is the pattern already exhaustive?

Reply:
Unreachable again as above

Reply:
In this case then remove the throw